cabal-install-solver/src/Distribution/Solver/Modular/Message.hs

```diff
@@ -41,51 +42,130 @@ data Message =
   | Success
   | Failure ConflictSet FailReason
 
+data Log
```

Comment:
It would help to have some comments describing how the new types and `Message` relate to each other.